pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -123,10 +126,15 @@ func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, upd
 	for _, c := range clusters {
 		builtClusters.Insert(c.Name)
 	}
-	finalDeletedClusters := slices.FilterInPlace(deletedClusters, func(cluster string) bool {
-		return !builtClusters.Contains(cluster)
-	})
-	return clusters, finalDeletedClusters, log, true
+	// Remove anything we built from the deleted list
+	deletedClusters = deletedClusters.Difference(builtClusters)
+	// Inbound is not handled above. Remove anything that we had before and no longer have
```

Comment:
why inbound is not handled above, iiuc onfiggen.buildClusters generate both inbound and outbound clusters

Reply:
Inbound deletes are not handled

Reply:
Can u explain why does not 	L119 deletedClusters contain inbound cluster

Reply:
Understood, it is hidden too deeper